src/server/scripts/DragonIsles/TheForbiddenReach/zone_the_war_creche.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "Containers.h"
+#include "Player.h"
+#include "Random.h"
+#include "Spell.h"
+#include "SpellScript.h"
+#include "Log.h"
+
+enum TheWarCreche
+{
+	// Spells
+
+	SPELL_DRACTHYR_LOGIN 							= 369728, // teleports to random room, plays scene for the room, binds the home position
+    SPELL_STASIS_1                                  = 369735, // triggers 366620
+    SPELL_STASIS_2                                  = 366620, // triggers 366636
+    SPELL_STASIS_3                                  = 366636, // removes 365560, sends first quest (64864)
+	SPELL_STASIS_4                                 	= 365560, // freeze the target
+	SPELL_DRACTHYR_EVOKER_LOGIN_POST_MOVIE_ROOM_01	= 394245, // scene for room 1
+	SPELL_DRACTHYR_EVOKER_LOGIN_POST_MOVIE_ROOM_02	= 394279, // scene for room 2
+	SPELL_DRACTHYR_EVOKER_LOGIN_POST_MOVIE_ROOM_03	= 394281, // scene for room 3
+	SPELL_DRACTHYR_EVOKER_LOGIN_POST_MOVIE_ROOM_04	= 394282, // scene for room 4
+	//SPELL_DRACTHYR_EVOKER_LOGIN_POST_MOVIE_ROOM_05	= 394283, // scene for room 5 (only plays sound)
+};
+
+typedef std::vector<std::pair<uint32, Position>> RoomPositionList;
```

Comment:
isnt what you need basically just a unordered map?

Reply:
no, its just required to select a random element from the container, map would be wrong use here

Reply:
thanks